apps/extension/src/routes/popup/settings/settings-connected-sites.tsx

```diff
@@ -0,0 +1,122 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Button, FadeTransition, Input } from '@penumbra-zone/ui';
+import { LinkGradientIcon } from '../../../icons';
+import { SettingsHeader } from '../../../shared';
+import { OriginRecord, localExtStorage } from '@penumbra-zone/storage';
+import { DisplayOriginURL } from '../../../shared/components/display-origin-url';
+import Map from '@penumbra-zone/polyfills/Map.groupBy';
+
+export const SettingsConnectedSites = () => {
+  const [search, setSearch] = useState<string>();
+
+  const [sitesFromStorage, setSitesFromStorage] = useState<OriginRecord[]>();
+
+  const loadSitesFromStorage = useCallback(
+    async () => setSitesFromStorage(await localExtStorage.get('connectedSites')),
+    [],
+  );
+  useEffect(() => void loadSitesFromStorage(), [loadSitesFromStorage]);
+
+  const discardSiteRecord = useCallback(
+    (site: OriginRecord) => {
+      if (!sitesFromStorage) return;
+      const sitesWithoutRecord = sitesFromStorage.filter(({ origin }) => origin !== site.origin);
+      void localExtStorage.set('connectedSites', sitesWithoutRecord);
+      setSitesFromStorage(sitesWithoutRecord);
+    },
+    [sitesFromStorage],
+  );
+
+  const filteredByAttitude = useMemo(
+    () =>
+      Map.groupBy(
+        (sitesFromStorage ?? []).filter(site => !search || site.origin.includes(search)),
+        ({ attitude }) => attitude,
+      ),
+    [sitesFromStorage, search],
+  );
+
+  return (
+    <FadeTransition>
+      <div className='flex min-h-[100vh] w-[100vw] flex-col gap-6'>
+        <SettingsHeader title='Connected sites' />
+        <div className='mx-auto size-20'>
+          <LinkGradientIcon />
+        </div>
+        <div className='flex flex-col gap-4 px-[30px]'>
+          <div className='relative flex w-full items-center justify-center gap-4'>
+            <div className='absolute inset-y-0 left-3 flex items-center'>
+              <MagnifyingGlassIcon className='size-5 text-muted-foreground' />
+            </div>
+            <Input
+              className='pl-10'
+              value={search}
+              onChange={e => setSearch(e.target.value)}
+              placeholder='Search by origin...'
+            />
+          </div>
+          <div className='flex flex-col gap-2'>
+            {filteredByAttitude
+              .get(true)
+              ?.map(site => (
+                <SiteRecord key={site.origin} site={site} discardSiteRecord={discardSiteRecord} />
+              ))
+              .toSpliced(0, 0, <div className='mt-2 text-muted-foreground'>Approved sites</div>)}
+            {filteredByAttitude
+              .get(false)
+              ?.map(site => (
+                <SiteRecord key={site.origin} site={site} discardSiteRecord={discardSiteRecord} />
+              ))
+              .toSpliced(0, 0, <div className='mt-2 text-muted-foreground'>Denied sites</div>)}
```

Comment:
.toSpliced() feels like an odd thing to find in a react component. It's not entirely clear what it's doing here. Can you re-write the template without it?